File: src-tauri/src/models/calendar.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Calendar {
    pub id: String,
    pub name: String,
    pub color: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub is_primary: bool,
}

impl Calendar {
    pub fn new(name: String, color: String) -> Self {
        let now = Utc::now();
        Self {
            id: uuid::Uuid::new_v4().to_string(),
            name,
            color,
            created_at: now,
            updated_at: now,
            is_primary: false,
        }
    }
}

File: src/types/calendar.ts
export interface Calendar {
  id: string;
  name: string;
  color: string;
  created_at: Date;
  updated_at: Date;
  is_primary: boolean;
}

export interface CalendarViewType {
  value: 'month' | 'week' | 'day';
  label: string;
}
